src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1030,7 +1031,7 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                     case (ASR::unaryopType::UAdd): { result = op_value; break; }
                     case (ASR::unaryopType::USub): { result = -op_value; break; }
                     default: {
-                        throw SemanticError("Unary operator not implemented yet for compile time evaluation",
+                        throw SemanticError("Bad operand type for unary operation",
```

Comment:
Also: I'm trying to improve the error messages. I just don't see a way to convert `ttypeType` to string, otherwise, we would have nice and self-explaining messages, something like this:
> "Bad operand type for unary operation ~: real"

Reply:
Okay, I found `type_to_str` inside ASR utils. That will work!

Reply:
Yes. And if you can't find what you need, just write it yourself and put it somewhere in `libasr`.